kotti/rest.py

```diff
@@ -0,0 +1,102 @@
+""" JSON Encoders, serializers, REST views and utilities
+"""
+
+from kotti.resources import Document
+from pyramid.renderers import JSONP
+from pyramid.view import view_config, view_defaults
+import colander
+import datetime
+import decimal
+import json
+
+
+def content_serializer(context, request):
+    from kotti.views.edit.content import ContentSchema
+    return ContentSchema().serialize(context.__dict__)
+
+
+def document_serializer(context, request):
+    from kotti.views.edit.content import DocumentSchema
+    return DocumentSchema().serialize(context.__dict__)
+
+
+def file_serializer(context, request):
+    from kotti.views.edit.content import FileSchema
+    return FileSchema(None).serialize(context.__dict__)
+
+
+ACCEPT = 'application/vnd.api+json'
+
+@view_defaults(name='json', accept=ACCEPT, renderer="kotti_jsonp")
+class RestView(object):
+    """ A generic @@json view for any and all contexts.
+
+    Its response depends on the HTTP verb used. For ex:
+    """
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    @view_config(request_method='GET')
+    def get(self):
+        return self.context
+
+    @view_config(request_method='POST')
+    def post(self):
+        pass
+
+    @view_config(request_method='PATCH')
+    def patch(self):
+        pass
+
+    @view_config(request_method='PUT')
+    def put(self):
+        pass
+
+    @view_config(request_method='DELETE')
+    def delete(self):
+        pass
+
+
+datetime_types = (datetime.time, datetime.date, datetime.datetime)
+
+def _encoder(basedefault):
+    """ A JSONEncoder that can encode some basic odd objects.
+
+    For most objects it will execute the basedefault function, which uses
+    adapter lookup mechanism to achieve the encoding, but for some basic
+    objects, such as datetime and colander.null we solve it here.
+    """
+
+    class Encoder(json.JSONEncoder):
+
+        def default(self, obj):
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [An attribute defined in json.encoder line 162 hides this method](https://www.codacy.com/app/disko/Kotti/file/941403419/issues/source?bid=2089747&fileBranchId=2722762#l74)